apps/studio/components/interfaces/Account/AccessTokens/NewAccessTokenButton.tsx

```diff
@@ -0,0 +1,242 @@
+import { zodResolver } from '@hookform/resolvers/zod'
+import { ChevronDown, ExternalLink } from 'lucide-react'
+import Link from 'next/link'
+import { useState } from 'react'
+import { type SubmitHandler, useForm } from 'react-hook-form'
+import { toast } from 'sonner'
+import { z } from 'zod'
+
+import { useAccessTokenCreateMutation } from 'data/access-tokens/access-tokens-create-mutation'
+import {
+  Button,
+  Sheet,
+  SheetContent,
+  SheetFooter,
+  SheetHeader,
+  SheetTitle,
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuItem,
+  DropdownMenuTrigger,
+  Form_Shadcn_,
+  ScrollArea,
+  Separator,
+} from 'ui'
+import { Admonition } from 'ui-patterns'
+
+import { TokenBasicInfoForm } from './TokenBasicInfoForm'
+import { TokenResourceAccessForm } from './TokenResourceAccessForm'
+import { TokenPermissionsForm } from './TokenPermissionsForm'
+
+export interface NewAccessTokenButtonProps {
+  onCreateToken: (token: any) => void
+}
+
+const PermissionRowSchema = z.object({
+  resource: z.string().min(1, 'Please select a resource'),
+  action: z.string().min(1, 'Please select an action'),
+})
+
+const TokenSchema = z.object({
+  tokenName: z.string().min(1, 'Please enter a name for the token'),
+  expirationDate: z.enum(['No expiry', '7 days', '30 days', '90 days', '180 days', 'Custom']),
+  resourceAccess: z.enum(['all-orgs', 'selected-orgs', 'selected-projects']),
+  selectedOrganizations: z.array(z.string()).optional(),
+  selectedProjects: z.array(z.string()).optional(),
+  organizationPermissions: z.record(z.string(), z.string()).optional(),
+  projectPermissions: z.record(z.string(), z.string()).optional(),
+  permissionRows: z.array(PermissionRowSchema).optional(),
+})
```

Comment:
@hieuwu appreciate the endpoint may not look like this, so let's adjust accordingly once that's ready!

Reply:
Cool. Great work. I am using some tricks in my code to handle different links, now with this fix I can standardize my code more. Thanks for the fix. Let's see 🤞